### ISSUE TITLE
Question: S4049 is raised if IAsyncEnumerable is returned

### ISSUE THREAD

**Issue by lg2de** [SonarSource/sonar-dotnet#3381]
### Description

S4049 is raised for the following interface method:
```cs
IAsyncEnumerable<SimpleEvent> GetEvents();
```

According to the documentation `async` methods are excluded.
Why methods with `IAsyncEnumerable` are not excluded?

### Related information

* C#/VB.NET Plugins Version: 8.6.1.17183

**Comment by costin-zaharia-sonarsource** [comment]
Hi @lg2de 

In order to be excluded the method should use the [async](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/async) modifier.

A few examples can be found here: https://docs.microsoft.com/en-us/archive/msdn-magazine/2019/november/csharp-iterating-with-async-enumerables-in-csharp-8

Is there any particular case you think we should consider when a method which returns `IAsyncEnumerable<T>` should not be decorated with `async` modifier?

**Comment by pavel-mikula-sonarsource** [comment]
Hi @lg2de ,

Situation here is similar to your [previous question](https://github.com/SonarSource/sonar-dotnet/issues/3264) about `IAsyncEnumerable`. `GetEvents()` itself is normal synchronous method that. It's result could be iterated with `async` on consumer side.

I'm not very sure about the "interface method" part of your question. According to the S4049 specification, interface implementations should be excluded. Can you clarify that or provide minimal reproducer for that?
